spec/system/admin/order_cycles/list_spec.rb

```diff
@@ -147,6 +147,32 @@
           expect(find('input.datetimepicker', match: :first).value).to eq '2012-01-30 00:00'
         end
       end
+
+      it "correctly opens the datetimepicker and closes it using the last button (the 'Close' one)" do
+        login_as_admin_and_visit admin_order_cycles_path
+        test_value = Time.parse("2022-12-22 00:00")
+
+        # Opens a datetimepicker
+        within("tr.order-cycle-#{oc_pt.id}") do
+          find('input.datetimepicker', match: :first).click
+        end
+
+        # Sets the value to test_value then looks for the close button and click it
+        within(".flatpickr-calendar.open") do
+          expect(page).to have_selector '.shortcut-buttons-flatpickr-buttons'
+          select_datetime_from_datepicker test_value
+          find(".shortcut-buttons-flatpickr-buttons > button:nth-last-child(1)").click
```

Comment:
It's better to test for what the user sees:
```suggestion
          find("button", text: "CLOSE").click
```